src/alarmHistory.ts

```diff
@@ -1,80 +1,104 @@
-import { DescribeAlarmHistoryOutput, AlarmHistoryItems } from "aws-sdk/clients/cloudwatch";
+import { AlarmHistoryItems, DescribeAlarmHistoryOutput } from "aws-sdk/clients/cloudwatch";
+import { CloudwatchStateChangeEvent } from './common';
 
 export enum AlarmState {
-  OK = 'OK',
-  ALARM = 'ALARM'
+  OK = "OK",
+  ALARM = "ALARM",
+  INSUFFICIENT_DATA = "INSUFFICIENT_DATA"
 }
 
-interface HistoryDataType {
-  version: '1.0',
+interface IHistoryDataType {
+  version: "1.0";
   oldState: {
     stateValue: string,
     stateReason: string,
-  },
+  };
   newState:
   {
     stateValue: string,
     stateReason: string,
     stateReasonData:
     {
-      version: '1.0',
+      version: "1.0",
       queryDate: string,
       startDate: string,
       statistic: string,
       period: number,
       recentDatapoints: number[],
-      threshold: number
-    }
-  }
+      threshold: number,
+    },
+  };
 }
 
-export function secondsBetween(alarmHistory: DescribeAlarmHistoryOutput, newState: AlarmState, oldState: AlarmState): number {
+export function secondsBetweenFromHistory(alarmHistory: DescribeAlarmHistoryOutput, newState: AlarmState, oldState: AlarmState):
+  number {
   let items = alarmHistory.AlarmHistoryItems || [];
   // const by = state => item => item.HistoryItemType === 'StateUpdate' && itemHasState(item.HistoryData, state);
 
-
   try {
-    const newStateDate = dateOfFirst(items, newState)
+    const newStateDate = dateOfFirst(items, newState);
     if (newState === oldState) {
-      items = removeFirst(items, newState)
+      items = removeFirst(items, newState);
     }
-    const oldStateDate = dateOfFirst(items, oldState)
+    const oldStateDate = dateOfFirst(items, oldState);
 
-    const diff = (newStateDate.getTime() - oldStateDate.getTime()) / 1000
+    const diff = (newStateDate.getTime() - oldStateDate.getTime()) / 1000;
 
     if (diff < 0) {
-      const msg = `States [${newState},${oldState}] found in the wrong order`
-      console.warn(msg)
-      const up = new Error(msg)
-      throw up
+      const msg = `States [${newState},${oldState}] found in the wrong order`;
+      console.warn(msg);
+      const up = new Error(msg);
+      throw up;
     }
 
-    return diff
+    return diff;
   } catch (err) {
-    throw new Error(`No alarms found in order: [${newState}, ${oldState}] in ${JSON.stringify(alarmHistory)}`)
+    throw new Error(`No alarms found in order: [${newState}, ${oldState}] in ${JSON.stringify(alarmHistory)}`);
   }
 }
 
-const by = state => item => item.HistoryItemType === 'StateUpdate' && itemHasState(item.HistoryData, state);
+export function secondsBetweenPreviouseState(event: CloudwatchStateChangeEvent, newState: AlarmState, oldState: AlarmState):
```

Comment:
newState / oldState are not used, other than for logging.

Reply:
Done, taken out